virttest/test_setup.py

```diff
@@ -376,7 +376,7 @@ def __init__(self, params):
         self.deallocate = params.get("hugepages_deallocate", "yes") == "yes"
         self.hugepage_path = '/mnt/kvm_hugepage'
         if os.path.exists('/proc/sys/vm/nr_hugepages'):
-            self.kernel_hp_file = '/proc/sys/vm/nr_hugepages'
+            self.kernel_hp_file = params.get("kernel_hp_file", "/proc/sys/vm/nr_hugepages")
```

Comment:
@dzhengfy on a lighter note, you misspelled my username :-)
I guess we can move this before the check for file, so that we can achieve it for user given path aswell.

how about 
```
self.kernel_hp_file = params.get("kernel_hp_file", "/proc/sys/vm/nr_hugepages")
if not os.path.exists(self.kernel_hp_file):
    raise exceptions.TestSkipError("System doesn't support hugepages,  (user given ) hugepage config file not found: %s" % self.kernel_hp_file)
```
something like this covers all cases and makes it informational for user, feel free to add/edit logging.

Reply:
@sathnaga , I used a simple error message which I think it is enough. Please help review again.